### PR TITLE
[clang-tidy] Ignore `if consteval` in else-after-return

### LINKED ISSUE #91561: `readability-else-after-return`/`llvm-else-after-return` warn on else after `if consteval`
The following yields the aforementioned warnings, but they shouldn't.
```cpp
constexpr int f()
{
  if consteval {
    return 0;
  }
  else {
    return 1;
  }
}
```

### PR DESCRIPTION
**Pull request by JoverZhang**
Fixes #91561.

### CODE REVIEW

**Comment by cjdb** [comment]
Thanks for your patch! Would you be able to add a test case for this please?

**Comment by JoverZhang** [comment]
> Thanks for your patch! Would you be able to add a test case for this please?

Yes, thanks. I added a test case.

File: clang-tools-extra/test/clang-tidy/checkers/readability/else-after-return-if-consteval.cpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// RUN: %check_clang_tidy -std=c++20 %s readability-else-after-return %t
```
Comment (5chmidti) [inline clang-tools-extra/test/clang-tidy/checkers/readability/else-after-return-if-consteval.cpp:1]: While `consteval` was added in C++20, `if consteval` is a C++23 feature. Please use `-std=c++23`
https://en.cppreference.com/w/cpp/language/if#Consteval_if

**Review by 5chmidti — COMMENTED** [review]
Please add an entry to the release notes here: https://github.com/llvm/llvm-project/blob/502e77df1fc4aa859db6709e14e93af6207e4dc4/clang-tools-extra/docs/ReleaseNotes.rst?plain=1#L176
do note that the entries are sorted by their check names.

Otherwise looks good.

**Comment by JoverZhang** [comment]
> Please add an entry to the release notes here:
> 
> https://github.com/llvm/llvm-project/blob/502e77df1fc4aa859db6709e14e93af6207e4dc4/clang-tools-extra/docs/ReleaseNotes.rst?plain=1#L176
> 
> 
> do note that the entries are sorted by their check names.
> Otherwise looks good.

Okay, I added.

**Review by HerrCai0907 — APPROVED** [review]

File: clang-tools-extra/docs/ReleaseNotes.rst [hunk 2]
```diff
@@ -337,6 +337,10 @@ Changes in existing checks
   <clang-tidy/checks/readability/duplicate-include>` check by excluding include
   directives that form the filename using macro.
 
+- Improved :doc:`readability-else-after-return
+  <clang-tidy/checks/readability/else-after-return>` check to ignore
+  `consteval if` condition, where need to retain the else statement.
```
Comment (SimplyDanny) [inline clang-tools-extra/docs/ReleaseNotes.rst:342]:
```suggestion
  `if consteval` statements, for which the `else` branch must not be removed.
```

**Review by SimplyDanny — APPROVED** [review]

**Review by 5chmidti — APPROVED** [review]

**Comment by SimplyDanny** [comment]
Thank you, @JoverZhang!

**Pull request merged**